crates/next-core/src/next_font/google/util.rs

```diff
@@ -205,17 +261,25 @@ pub(super) fn get_stylesheet_url(
                 .map(|variant| {
                     variant
                         .iter()
-                        .map(|pair| pair.1)
-                        .collect::<Vec<&str>>()
-                        .join(",")
+                        .map(|pair| pair.1.clone())
```

Comment:
@bgw do you have an idea of how I can accomplish this with less cloning? If `variants` is a `Vec<Vec<(&str, &VariantValue)>>` instead of `Vec<Vec<(&str, VariantValue)>>`, the iterator yields `&&VariantValue`, which can't be dereferenced multiple times to stringify without upsetting the borrow checker.

Reply:
The cloning is necessary a something has to perform an incref on the inner `RcStr`, and the `impl From<VariantValue> for RcStr` doesn't do that.

I don't think it's a huge deal since it's just a refcount increment in the worst-case. Technically this could be worked around by exposing an API that returns a `Cow<'_, str>` instead of `RcStr`, but I think that'd be overkill.

However, I would defer the cloning until the last possible moment, i.e.

```
            let mut variant_values = variants
                .iter()
                .map(|variant| variant.iter().map(|pair| &pair.1).collect::<Vec<_>>())
                .collect::<Vec<Vec<_>>>();
            variant_values.sort();

            // An encoding of the series of sorted variant values, with variants delimited
            // by `;` and the values within a variant delimited by `,` e.g.
            // `"0,10..100,500;1,10.100;500"`
            let variant_values_str = variant_values
                .iter()
                .map(|v| {
                    v.iter()
                        .map(|vv| RcStr::from((*vv).clone()))
                        .collect::<Vec<_>>()
                        .join(",")
                })
                .collect::<Vec<_>>()
                .join(";");
```

Additional notes:

- `RcStr::from` is equivalent to `std::convert::Into::<RcStr>::into` and much simpler
- Use the `_` placeholder type where possible to keep the code a bit more compact (though it's less explicit, so that's a matter of opinion).